File: 0264-ugly-number-ii/0264-ugly-number-ii.cpp
class Solution {
public:
    int nthUglyNumber(int n) {
        int a = 2;
        int b = 3;
        int c = 5;
        int k = 1,l=1,m=1;
        vector<int> res;
        res.push_back(1);
        for(int i=1;i<n;i++)
        {
            int small = min(a,min(b,c));
            res.push_back(small);
            if(small==a)
            {
                a = res[k]*2;
                k++;
            }
            if(small==b)
            {
                b = res[l++]*3;
            }
            if(small==c)
            {
                c = res[m++]*5;
            }
        }
        return res[n-1];
    }
};

File: 0264-ugly-number-ii/0264-ugly-number-ii.py
class Solution:
    def nthUglyNumber(self, l: int) -> int:
        res = []
        res.append(1)
        m,n,o=1,1,1
        a,b,c=2,3,5
        for i in range(1,l):
            curr = min(a,min(b,c))
            res.append(curr)
        
            if curr==a:
                a = res[m]*2
                m+=1
            if curr==b:
                b = res[n]*3
                n+=1
            if curr==c:
                c = res[o]*5
                o+=1

        return res[-1]
            


        